apps/wallet/src/ui/app/pages/home/tokens/coin-balance/index.tsx

```diff
@@ -1,46 +1,56 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useFormatCoin } from '@mysten/core';
-import cl from 'classnames';
-import { memo, useCallback } from 'react';
-import { useNavigate } from 'react-router-dom';
-
-import { CoinItem } from '_components/active-coins-card/CoinItem';
-
-import st from './CoinBalance.module.scss';
+import { useFormatCoin, useSuiCoinData } from '@mysten/core';
+import { SUI_DECIMALS } from '@mysten/sui.js/utils';
+import BigNumber from 'bignumber.js';
+import { useMemo } from 'react';
+import { CoinIcon } from '_src/ui/app/components/coin-icon';
+import { Heading } from '_src/ui/app/shared/heading';
+import { Text } from '_src/ui/app/shared/text';
 
 export type CoinProps = {
 	type: string;
-	balance: bigint;
-	hideStake?: boolean;
-	mode?: 'row-item' | 'standalone';
+	amount: bigint;
 };
 
-function CoinBalance({ type, balance, mode = 'row-item' }: CoinProps) {
-	const [formatted, symbol] = useFormatCoin(balance, type);
-	const navigate = useNavigate();
+export function CoinBalance({ amount: walletBalance, type }: CoinProps) {
+	const [formatted, symbol] = useFormatCoin(walletBalance, type);
+	const { data } = useSuiCoinData();
```

Comment:
remind me to up our CoinGecko plan before we ship this to users... that and we need to figure out some sensible cache times for this query 😅